lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -2826,4 +2841,60 @@ void SessionState::process_get_5g_rule_installs(
   }
 }
 
+void SessionState::process_get_mod_rule_installs(
+    const std::vector<QosPolicy>& qospolicy, RulesToProcess* pending_activation,
+    RulesToProcess* pending_deactivation) {
+  for (const QosPolicy& qos_policy : qospolicy) {
+    const std::string& rule_id = qos_policy.qos().id();
+    if (is_static_rule_installed(rule_id)) {
+      PolicyRule rule;
+      RuleToProcess to_process;
+      static_rules_.get_rule(rule_id, &rule);
+      to_process.version = get_current_rule_version(rule.id());
+      to_process.rule = rule;
+      to_process.teids = config_.common_context.teids();
+      if (qos_policy.policy_action() == QosPolicy::ADD) {
+        MLOG(MDEBUG) << " Static Rule pending_activation qos_policy.version :"
+                     << qos_policy.version()
+                     << " to_process.version :" << to_process.version;
+        if (qos_policy.version() == to_process.version) {
+          pending_activation->push_back(to_process);
+        }
+      } else if (qos_policy.policy_action() == QosPolicy::DEL) {
+        if (qos_policy.version() == to_process.version) {
+          MLOG(MDEBUG)
+              << " Static Rule pending_deactivation qos_policy.version :"
+              << qos_policy.version()
+              << " to_process.version :" << to_process.version;
+          pending_deactivation->push_back(to_process);
+        }
+      }
+    } else {
+      PolicyRule rule;
+      RuleToProcess to_process;
+      dynamic_rules_.get_rule(rule_id, &rule);
+      to_process.version = get_current_rule_version(rule.id());
+      to_process.rule = rule;
+      to_process.teids = config_.common_context.teids();
+      if (qos_policy.policy_action() == QosPolicy::ADD) {
+        if (qos_policy.version() == to_process.version) {
+          MLOG(MDEBUG)
+              << " Dynamic Rule pending_activation qos_policy.version :"
+              << qos_policy.version()
+              << " to_process.version :" << to_process.version;
+          pending_activation->push_back(to_process);
+        }
+      } else if (qos_policy.policy_action() == QosPolicy::DEL) {
+        if (qos_policy.version() == to_process.version) {
```

Comment:
do you really need another if here? Can't be the condition added  to the first if?

Reply:
Yeah added condition as suggested instead of one more if statement.